### PR TITLE
[WFLY-14271] timed object id throws null when tried to be restored

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/WFLY-14271
Upon activation of the timer service the creation of the hash set related to a timer object id can be null if there is a problem with database connection. This causes the map not to be filled and therfore any further registration of the timers fail to NullPointerException.
Added a new timerDeployed event in the persistence to signal before the load so always this is filled.

### CODE REVIEW

**Comment by elguardian** [comment]
this is ok to test

File: ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java:332]: you need to add the 'throw' keyword.

Comment (bstansberry) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java:332]: BTW I had a quick look because the PR title made me curious, and noted the ^^^, but please don't wait on me for approval on this; the EJB experts should do the code review.

Comment (elguardian) [reply ejb3/src/main/java/org/jboss/as/ejb3/timerservice/persistence/database/DatabaseTimerPersistence.java:332]: thx @bstansberry for the feedback. Sry I was testing CI in this case for not breaking anything before asking for Cheng

**Comment by elguardian** [comment]
@chengfang do you mind to have a look. We have big customer on this.

@bstansberry cannot add reviewers

**Comment by elguardian** [comment]
Jenkins retest this please

**Comment by elguardian** [comment]
This is ok to test

**Comment by bstansberry** [comment]
The github request review function is not very useful as there are far more people who should do reviews vs ones who can hit the merge button. Using @ pings works better

@tadamski may be able to help as well

**Review by chengfang — APPROVED** [review]

**Pull request merged**
